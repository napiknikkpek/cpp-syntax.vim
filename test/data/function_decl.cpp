void foo(Foo a);
void foo(Foo&& a);
void foo(const Foo a)
void foo(const Foo& a);
void foo(Foo& a);
void foo(Foo const& a);
void foo(Foo const* const a);
void foo(int a, char);
void* foo(int a, char);
void const* foo(int a, char);
void const* const foo(int a, char);
Item foo(Foo, Bar a);
Item& foo(Foo, Bar a);
Item const& foo(Foo, Bar a);
Item const& const foo(Foo, Bar a);
Item<X> foo(Foo, Bar a);
Item<X>* foo(Foo, Bar a);
Item<X> const* foo(Foo, Bar a);
Item<X> const* const foo(Foo, Bar a);
Item::A foo(Foo, Bar a);
Item::A A::foo(Foo, Bar a);
Item A::foo<int, A>(Foo, Bar a);
Item<X> Item::foo<int, A>(Foo, Bar a);
Item<X>& Item::foo<int, A>(Foo, Bar a);
Item<X> const& Item::foo<int, A>(Foo, Bar a);
Item<X> const& const Item::foo<int, A>(Foo, Bar a);
const A foo(Foo, Bar a);
A<X>&& foo(Foo, Bar a);
