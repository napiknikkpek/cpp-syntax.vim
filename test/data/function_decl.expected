zzzz.fff.ttt.x..
zzzz.fff.ttt...x..
zzzz.fff.aaaaa.ttt.x.
zzzz.fff.aaaaa.ttt..x..
zzzz.fff.ttt..x..
zzzz.fff.ttt.aaaaa..x..
zzzz.fff.ttt.aaaaa..aaaaa.x..
zzzz.fff.zzz.x..zzzz..
zzzz..fff.zzz.x..zzzz..
zzzz.aaaaa..fff.zzz.x..zzzz..
zzzz.aaaaa..aaaaa.fff.zzz.x..zzzz..
tttt.fff.ttt..ttt.x..
tttt..fff.ttt..ttt.x..
tttt.aaaaa..fff.ttt..ttt.x..
tttt.aaaaa..aaaaa.fff.ttt..ttt.x..
tttt.t..fff.ttt..ttt.x..
tttt.t...fff.ttt..ttt.x..
tttt.t..aaaaa..fff.ttt..ttt.x..
tttt.t..aaaaa..aaaaa.fff.ttt..ttt.x..
tttt..t.fff.ttt..ttt.x..
tttt..t.t..fff.ttt..ttt.x..
tttt.t..fff.zzz..t..ttt..ttt.x..
tttt.t..tttt..fff.zzz..t..ttt..ttt.x..
tttt.t...tttt..fff.zzz..t..ttt..ttt.x..
tttt.t..aaaaa..tttt..fff.zzz..t..ttt..ttt.x..
tttt.t..aaaaa..aaaaa.tttt..fff.zzz..t..ttt..ttt.x..
aaaaa.t.fff.ttt..ttt.x..
t.t....fff.ttt..ttt.x..
